feat(actors) Add user_id and team_id columns to Rule

markstory opened the pull request:
Part of the `Actor` removal. Add user_id and team_id columns to `Rule`. These columns will replace the `owner_id` column.

Refs HC-1168

markstory committed:
feat(actors) Add user_id and team_id columns to Rule

Part of the `Actor` removal. Add user_id and team_id columns to `Rule`.
These columns will replace the `owner_id` column.

Refs HC-1168

corps commented:
You'll need to disable or fix migration test test_0670_monitor_incident_cleanup_duplicates

wedamija commented on src/sentry/models/rule.py:
```diff
@@ -57,6 +58,8 @@ class Rule(Model):
         choices=RuleSource.as_choices(),
     )
     owner = FlexibleForeignKey("sentry.Actor", null=True, on_delete=models.SET_NULL)
+    user_id = HybridCloudForeignKey(settings.AUTH_USER_MODEL, null=True, on_delete="SET_NULL")
+    team = FlexibleForeignKey("sentry.Team", null=True, on_delete=models.SET_NULL)
```
I think these should be `owner_user_id` and `owner_team_id`.

We should also add a constraint to guarantee that only one is set.

markstory replied:
Sounds good, I've made those changes now.

markstory committed:
Remove migration temporarily

markstory committed:
Merge branch 'master' into feat-team-user-rule

markstory committed:
Rename columns and add check constraint.

markstory committed:
Remove migration tests that are now broken

These tests are all broken because of the new columns. The new columns
are retained in state, but are no longer in the schema because
migrations are rolled back.

markstory committed:
Update snapshots
